app/src/main/java/ar/com/wolox/android/cookbook/mpchart/MpChartRecipeFragment.kt

```diff
@@ -0,0 +1,356 @@
+            visibility = View.VISIBLE
+            data = bubbleData
+            isDragEnabled = true
+            description.text = getString(R.string.mp_chart_bubble)
+
+            animateXY(ANIMATION_DELAY, ANIMATION_DELAY)
+
+            setOnChartValueSelectedListener(object : OnChartValueSelectedListener {
+                override fun onValueSelected(e: Entry?, h: Highlight?) {
+                    val entry = e as BubbleEntry
+
+                    /**
+                     * Data is an additional information of Entry, represents an Object with
+                     * any class or null if no data has been specified
+                     */
+                    Toast.makeText(context, getString(R.string.mp_chart_item_bubble, entry.x, entry.y,
+                            entry.size, entry.data), Toast.LENGTH_SHORT).show()
+                }
+
+                override fun onNothingSelected() {
+                }
+            })
+        }
+    }
+
+    override fun showCombinedChart(combinedData: CombinedData) {
+        vCombinedChart.apply {
+            visibility = View.VISIBLE
+            data = combinedData
+
+            setDrawBarShadow(false)
+            description.text = getString(R.string.mp_chart_combined)
+            setDrawGridBackground(false)
+            isHighlightFullBarEnabled = false
+
+            axisRight.apply {
+                setDrawGridLines(false)
+                axisMinimum = ZERO
+            }
+
+            axisLeft.apply {
+                setDrawGridLines(false)
+                axisMinimum = ZERO
+            }
+
+            xAxis.apply {
+                position = XAxis.XAxisPosition.BOTH_SIDED
+                axisMinimum = ZERO
+                granularity = GRANULARITY
+                axisMaximum = data.xMax + AXIS_PHASE
+            }
+
+            animateXY(ANIMATION_DELAY, ANIMATION_DELAY)
+            invalidate()
+        }
+    }
+
+    override fun showHorizontalBarChart(hBarData: BarData) {
+        vHorizontalBarChart.apply {
+            visibility = View.VISIBLE
+            data = hBarData
+            animateY(ANIMATION_DELAY)
+            isDragEnabled = true
+
+            axisLeft.mAxisMinimum = ZERO
+
+            setOnChartValueSelectedListener(object : OnChartValueSelectedListener {
+                override fun onValueSelected(e: Entry?, h: Highlight?) {
+                    val entry = e as BarEntry
+
+                    /**
+                     * Data is an additional information of Entry, represents an Object with
+                     * any class or null if no data has been specified
+                     */
+                    Toast.makeText(context, getString(R.string.mp_chart_item_bar, entry.y, entry.data),
+                            Toast.LENGTH_SHORT).show()
+                }
+
+                override fun onNothingSelected() {
+                }
+            })
+
+            xAxis.apply {
+                mAxisMinimum = ZERO
```

Comment:
You don't need to define a constant for a number. The constants are util for making more understandable our code.